src/clientlayers/ConnectionRequest.jl

```diff
@@ -73,10 +73,17 @@ function connectionlayer(handler)
             url = target_url = req.url
         end
 
+        connect_timeout = connect_timeout == 0 && readtimeout > 0 ? readtimeout : connect_timeout
         IOType = sockettype(url, socket_type, socket_type_tls)
         start_time = time()
         try
-            io = newconnection(IOType, url.host, url.port; readtimeout=readtimeout, kw...)
+            io = if connect_timeout > 0
+                try_with_timeout(connect_timeout) do _
```

Comment:
Are we potentially using 2 Timers on a single `newconnection`? I think `newconnection` can call `getconnection` which also does `try_with_timeout` and has a default value `connect_timeout::Int=10`.